src/Servers/Kestrel/Core/src/ListenOptionsHttpsExtensions.cs

```diff
@@ -201,15 +193,31 @@ internal static bool TryUseHttps(this ListenOptions listenOptions)
     /// <returns>The <see cref="ListenOptions"/>.</returns>
     public static ListenOptions UseHttps(this ListenOptions listenOptions, HttpsConnectionAdapterOptions httpsOptions)
     {
-        var loggerFactory = listenOptions.KestrelServerOptions?.ApplicationServices.GetRequiredService<ILoggerFactory>() ?? NullLoggerFactory.Instance;
+        return listenOptions.UseHttps(new Lazy<HttpsConnectionAdapterOptions>(() =>
+        {
+            httpsOptions.HttpProtocols = listenOptions.Protocols;
```

Comment:
Putting this assignment inside the Lazy callback still feels wrong. It seems like it'd be super easy to forget. It was easy to forget before, but at least it was only set in one place instead of two.

Let's add an HttpProtocols parameter to the internal HttpsConnectionMiddleware, pass it directly from ListenOptions, and delete HttpsConnectionAdapterOptions.HttpProtocols.

Reply:
Always happy to store something in fewer places. :smile: